src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIApiAlertCodeFixer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    [ApplicableComponents(ProjectComponents.WinUI)]
+    public class WinUIApiAlertCodeFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(
+            "UA306_A1",
+            "UA306_A2",
+            "UA306_A3",
+            "UA306_A4",
+            "UA306_B",
+            "UA306_C",
+            "UA306_D",
+            "UA306_E",
+            "UA306_F",
+            "UA306_G",
+            "UA306_H",
+            "UA306_I");
+
+        private readonly ILogger _logger;
+
+        public WinUIApiAlertCodeFixer(ILogger<WinUIApiAlertCodeFixer> logger)
```

Comment:
codefixers should not be taking constructor args. If you want to run it in VS, then it will not work like this

Reply:
But will these ever run in VS?
And what's the alternative to using the logger if not this way?